src/raftstore/store/store.rs

```diff
@@ -1808,13 +1808,18 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         self.register_merge_check_tick(event_loop);
     }
 
-    fn on_ready_prepare_merge(&mut self, region: metapb::Region, state: MergeState) {
+    fn on_ready_prepare_merge(&mut self, region: metapb::Region, state: MergeState, merged: bool) {
         {
             let peer = self.region_peers.get_mut(&region.get_id()).unwrap();
             peer.pending_merge = Some(state);
             peer.mut_store().region = region.clone();
         }
 
+        if merged {
+            // PrepareMerge is executed by CommitMerge.
```

Comment:
This comment confuses me, what is "PrepareMerge is executed"?

Reply:
Just as its literal meaning. `CommitMerge` will try to catch up log for source region, which may execute `PrepareMerge`.

Reply:
Suggest adding more comments here.